### PR TITLE
Improved replace_special function

### PR DESCRIPTION
**Pull request by OksanaTk**
New version of replace_special, takes string of math mode as input, and only runs that string through the remove_speciall function.

### CODE REVIEW

**Comment by plato2000** [comment]
@physikerwelt Why is travis not running here? It should fail on the unit tests but it's not running them.

**Comment by plato2000** [comment]
@physikerwelt never mind, fixed it

**Comment by physikerwelt** [comment]
Can you change the title of the pull request, please?

**Comment by physikerwelt** [comment]
for the ci problems you can click on "Details" and the problem will be shown to you

File: AlexDanoff/src/math_function.py [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-from utilities import readin
-from utilities import writeout
 import re
 
 def math_string(file):
     # Takes input file and returns list of strings when in math mode
     string = open(file).read()
     output = []
     ranges = math_mode.find_math_ranges(string)
+    #print ranges
```
Comment (physikerwelt) [inline AlexDanoff/src/math_function.py]: you should not comment out code.

File: AlexDanoff/src/math_function.py [hunk 2]
```diff
@@ -27,46 +26,87 @@ def change_original(o_file, changed_math_string):
...
-def index_spacing(file, out_file):
-    used = open(file).read()
+def formatting(file_str):
+    # Proper spacing for the indexes and gets rid of all non begin{eq end{eq text
+
+    commands = r'\\[a-zA-Z]+'
 
     updated = []
-    IND_START = r'\index{'
+    IND_START = r'\index\{'
```
Comment (physikerwelt) [inline AlexDanoff/src/math_function.py]: why not \\index\{?

File: AlexDanoff/src/math_function.py [hunk 3]
```diff
@@ -27,46 +26,87 @@ def change_original(o_file, changed_math_string):
...
 
             fullsplit = ind_str.split("\n")
-            updated.extend(fullsplit)
             ind_str = ""
+
         previous = line
-        updated.append(line)
 
     wrote = "\n".join(updated)
+    wrote = wrote
```
Comment (physikerwelt) [inline AlexDanoff/src/math_function.py]: remove

File: AlexDanoff/src/math_function.py [hunk 4]
```diff
@@ -27,46 +26,87 @@ def change_original(o_file, changed_math_string):
...
     wrote = re.sub(r'\\index{(.*?)}\n\n\\index{(.*?)}', r'\\index{\1}\n\\index{\2}', wrote)
 
-    out = open(out_file, 'w')
-    out.write(wrote)
-    out.close()
-    return out
-# 4/26, look at chapter 16 and prevent converting the \ApellFiii into \ApellFii\iunit
+    return wrote
 
+#formatting(open('/home/ont1/DLMF/25.ZE/newMoritz').read())
```
Comment (physikerwelt) [inline AlexDanoff/src/math_function.py]: remove line
File should end with \n

File: AlexDanoff/src/replace_special.py [hunk 5]
```diff
@@ -210,30 +192,34 @@ def remove_special(content):
         function = re.sub(r'\\index{(.*?)}\n\n\\index{(.*?)}', r'\\index{\1}\n\\index{\2}', function)
 
         content[counter] = function
+        # print 'content[counter]', content[counter]
```
Comment (physikerwelt) [inline AlexDanoff/src/replace_special.py]: remove

File: AlexDanoff/src/math_function.py (hunk 2 above)
Comment (OksanaTk) [inline AlexDanoff/src/math_function.py]: { is a special character in regex. However, this time the code does the same thing.

**Pull request merged**